int32 samples           # Number of samples used for computing
                        # the error
float64 last_error      # Last error
float64 average_error   # Average error
